src/Model/Table/QueueProcessesTable.php

```diff
@@ -59,6 +59,19 @@ public function initialize(array $config): void {
 		$this->setPrimaryKey('id');
 
 		$this->addBehavior('Timestamp');
+
+		$this->hasOne('CurrentQueuedJobs', [
+				'className' => 'Queue.QueuedJobs',
+				'foreignKey' => 'workerkey',
+				'bindingKey' => 'workerkey',
+				'propertyName' => 'active_job',
+				'conditions' => [
+					'CurrentQueuedJobs.completed IS NULL',
+					'CurrentQueuedJobs.failure_message IS NULL',
+					'CurrentQueuedJobs.failed = 0',
```

Comment:
Why are those relevant?
it sometimes can be failed 1 and still being re-running.
So I wonder if we could just sync over workerkey config alone?

Reply:
right, I didn't think of failed jobs which can be re-run.